tendrl/gluster_integration/message/gluster_native_message_handler.py

```diff
@@ -0,0 +1,97 @@
+from flask import Flask
+from flask import request
+import gevent
+import gevent.event
+import gevent.greenlet
+from gevent.wsgi import WSGIServer
+
+from tendrl.commons.utils import cmd_utils
+from tendrl.commons.utils import log_utils as logger
+from tendrl.commons.utils import service as svc
+from tendrl.gluster_integration.message import callback as cb
+
+
+app = Flask(__name__)
+
+
+class GlusterNativeMessageHandler(gevent.greenlet.Greenlet):
+    def __init__(self):
+        super(GlusterNativeMessageHandler, self).__init__()
+        self.path = "/listen"
+        self.host = "0.0.0.0"
+        self.port = 9000
+        self.callback = cb.Callback()
+
+        @app.route(self.path, methods=["POST"])
+        def events_listener():
+            gluster_event = request.json
+            if gluster_event:
+                callback_function_name = gluster_event["event"].lower()
+                try:
+                    function = getattr(self.callback, callback_function_name)
+                except AttributeError:
+                    # tendrl does not handle this perticular event hence ignore
+                    pass
+                event_handler = gevent.spawn(function, gluster_event)
+                event_handler.join()
+
+    def _setup_gluster_native_message_reciever(self):
+        service = svc.Service("glustereventsd")
+        message, success = service.start()
```

Comment:
this service "glustereventsd" can be started by Storage admin too,  first check if the service is active, if no then try to start

Reply:
ack